src/components/AnnotationsCarousel/AnnotationsCarousel.tsx

```diff
@@ -0,0 +1,85 @@
+import { Block, Elem } from '../../utils/bem';
+import { LsChevron } from '../../assets/icons';
+import { Button } from '../../common/Button/Button';
+import './AnnotationsCarousel.styl';
+import { AnnotationButton } from './AnnotationButton';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { observer } from 'mobx-react';
+import { clamp } from '../../utils/utilities';
+
+interface AnnotationsCarouselInterface {
+  store: any;
+  annotationStore: any; 
+  commentStore?: any;
+}
+
+export const AnnotationsCarousel = observer(({ store, annotationStore }: AnnotationsCarouselInterface) => {
+  const [entities, setEntities] = useState<any[]>([]);
+  const enableAnnotations = store.hasInterface('annotations:tabs');
+  const enablePredictions = store.hasInterface('predictions:tabs');
+  const enableCreateAnnotation = store.hasInterface('annotations:add-new');
+  const groundTruthEnabled = store.hasInterface('ground-truth');
+  const enableAnnotationDelete = store.hasInterface('annotations:delete');
+  const carouselRef = useRef<HTMLElement>();
+  const containerRef = useRef<HTMLElement>();
+  const [currentPosition, setCurrentPosition] = useState(0);
+  const [isLeftDisabled, setIsLeftDisabled] = useState(false);
+  const [isRightDisabled, setIsRightDisabled] = useState(false);
+
+  const updatePosition = useCallback((e: MouseEvent, goLeft = true) => {
+    if (containerRef.current && carouselRef.current) {
+      const step = containerRef.current.clientWidth;
+      const carouselWidth = carouselRef.current.clientWidth;
+      const newPos = clamp(goLeft ? currentPosition - step : currentPosition + step, 0, carouselWidth - step);
+
+      setCurrentPosition(newPos);
+    }
+  }, [containerRef, carouselRef, currentPosition]);
+  
+  useEffect(() => {
+    setIsLeftDisabled(currentPosition <= 0);
+    setIsRightDisabled(currentPosition >= ((carouselRef.current?.clientWidth ?? 0) - (containerRef.current?.clientWidth ?? 0)));
+  }, [containerRef.current, carouselRef.current, currentPosition, window.innerWidth, window.innerHeight]);
```

Comment:
@Travis1282 We need to update this useEffect as well with the change in `entities.length`. Otherwise it falls into a case where duplicating/removing annotations it won't correctly update the visibility of the carousel buttons.

Reply:
its kinda of strange - i tested this exact case but it appears to be ok on my screen (< 1800 px wide) but the issue is easily seen on a wider screen (> 2000px wide)